cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -707,6 +728,24 @@ private String cleanse(String val, int maxLength) {
     return cleansed.length() > maxLength ? cleansed.substring(0, maxLength) : cleansed;
   }
 
+  /**
+   * Creates per-run configmap containing config files.
+   */
+  private void createConfigMap() throws ApiException, IOException {
+    String configMapName = CONFIGMAP_NAME_PREFIX + twillRunId.getId();
+    V1ConfigMapBuilder builder = new V1ConfigMapBuilder().withMetadata(new V1ObjectMeta().name(configMapName));
+    String runnableName = getMainRuntimeSpecification(twillSpec.getRunnables()).getName();
+    for (LocalFile localFile : twillSpec.getRunnables().get(runnableName).getLocalFiles()) {
+      // Only add xml config files to the configmap. We skip over jars as they are too large store in a
+      // configmap (max size = 1 MB. See https://kubernetes.io/docs/concepts/configuration/configmap).
+      if (!localFile.getName().endsWith(".xml")) {
+        continue;
+      }
+      builder.addToBinaryData(localFile.getName(), Files.readAllBytes(java.nio.file.Paths.get(localFile.getURI())));
+    }
+    coreV1Api.createNamespacedConfigMap(programRuntimeNamespace, builder.build(), null, null, null);
```

Comment:
should add the same labels to the configmap that we do for the other resources of the run. I seem to remember we have labels for the twill runid and other things

Reply:
added labels

Reply:
Labels added: cdap.instance, cdap.twill.app, cdap.twill.run.id, cdap.twill.runner